lib/apartment/adapters/abstract_adapter.rb

```diff
@@ -163,9 +181,9 @@ def create_tenant(tenant)
       def connect_to_new(tenant)
         Apartment.establish_connection multi_tenantify(tenant)
         Apartment.connection.active?   # call active? to manually check if this connection is valid
-
-      rescue *rescuable_exceptions
-        raise TenantNotFound, "The tenant #{environmentify(tenant)} cannot be found."
+      rescue *rescuable_exceptions => exception
+        Apartment::Tenant.reset if reset_on_connection_exception?
+        connect_to_tenant_error(tenant, exception)
```

Comment:
I was about to say this is a change in behaviour, but I think it's just the method naming, can you actually call it something that indicates we're raising, like `raise_connect_error!` or something?